src/Lecture2.hs

```diff
@@ -85,7 +106,16 @@ lists of even lengths.
 ♫ NOTE: Use eta-reduction and function composition (the dot (.) operator)
   in this function.
 -}
-evenLists = error "TODO"
+evenLists :: [[a]] -> [[a]]
+evenLists [] = []
+evenLists [[]] = [[]]
+evenLists (x:xs) = filterEven x ++ evenLists xs
```

Comment:
You're correctly using `even` and `length` function to check all lists 🙂 

But additionally you can just use a standard function `filter` with this check to implement `evenLists` in a single line ➡️